BUG: SARIMAX dates if simple_differencing=True

ChadFulton opened the pull request:
Fixes the problem with incorrect prediction index - and more generally with incorrect values in `self.data.dates` - when `simple_differencing=True`.

See #2939 for the bug report.

josef-pkt commented on statsmodels/tsa/statespace/tests/test_prediction.py:
```diff
@@ -0,0 +1,47 @@
+    assert_equal(pred.index.values, index[1:-1].values)
+    # Out-of-sample forecasting should still extend the index appropriately
+    fcast = res.forecast()
+    assert_equal(fcast.index[0], index[-1])
```
minor formatting issue. git/python wants a module to end with a newline

josef-pkt commented:
looks good to me, merge?

ChadFulton commented:
I added another test where I use seasonal differencing also, and added a newline. It should pass, and if it does it's good to merge.

Thanks @TomAugspurger
